### PR TITLE
Remove deprecated args and class for v1.3

### LINKED ISSUE #7018: Remove deprecated args and class for v1.3
Remove deprecated args and class for v1.3.

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #7018.

### Description
- Remove `compute_meaniou`
- Remove `AsChannelFirst`, `AddChannel` and `SplitChannel`
- Remove `create_multigpu_supervised_trainer` and `create_multigpu_supervised_evaluator`
- deprecated `data_src_cfg_filename` in `AlgoEnsembleBuilder`
- deprecated `runner_id` in `run`
- deprecated `epoch_interval` and `iteration_interval` in `TensorBoardStatsHandler`
- deprecated `get_validation_stats` in `Evaluator`
- deprecated `get_train_stats` in `Trainer`
- update the default value of `name`  to `StatsHandler` in `StatsHandler`
- update the default value of `random_size` to `False` in `RandSpatialCrop`, `RandScaleCrop`, `RandSpatialCropSamples`
- update the default value of `image_only` to `True` in `LoadImage` and `SaveImage`
- update `MetaTenor` related deprecation

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/handlers/tensorboard_handlers.py [hunk 1]
```diff
@@ -87,18 +87,14 @@ class TensorBoardStatsHandler(TensorBoardHandler):
...
     """
 
-    @deprecated_arg("epoch_interval", since="1.1", removed="1.3")
-    @deprecated_arg("iteration_interval", since="1.1", removed="1.3")
     def __init__(
         self,
         summary_writer: SummaryWriter | SummaryWriterX | None = None,
         log_dir: str = "./runs",
         iteration_log: bool | Callable[[Engine, int], bool] = True,
         epoch_log: bool | Callable[[Engine, int], bool] = True,
```
Comment (KumoLiu) [inline monai/handlers/tensorboard_handlers.py:95]: Looks like after removing `epoch_log` and `iteration_log` can also support `int`.

Comment (ericspod) [reply monai/handlers/tensorboard_handlers.py:95]: If this is something to add I'd suggest raising an issue about it and keep this PR on the topic of removing things only.

Comment (KumoLiu) [reply monai/handlers/tensorboard_handlers.py:95]: Sure, will do that, thanks!

**Comment by wyli** [comment]
thanks, we may consider updating the default value here: https://github.com/Project-MONAI/MONAI/blob/f3200b999c6332e7764ee8f71f7bd0f69ed2e2a7/monai/transforms/io/array.py#L130

File: monai/handlers/tensorboard_handlers.py (hunk 1 above)
Comment (wyli) [reply monai/handlers/tensorboard_handlers.py:95]: the `epoch_log`, `iteration_log` docstring currently don't state this feature clearly, the input could be boolean/int/callable...

File: monai/transforms/croppad/array.py [hunk 2]
```diff
@@ -819,13 +815,12 @@ def threshold_at_one(x):
 
     """
 
-    @deprecated_arg_default("allow_smaller", old_default=True, new_default=False, since="1.2", replaced="1.3")
```
Comment (wyli) [inline monai/transforms/croppad/array.py:822]: this is change of default was introduced after tagging v1.2, I think this warning should be kept for a few more versions

Comment (KumoLiu) [reply monai/transforms/croppad/array.py:822]: Sure, how about replacing it after "1.5"?

**Review by wyli — APPROVED** [review]
thanks, it looks good to me, running more tests then merging

**Pull request merged**